force-app/main/default/classes/DataWeaveInApex Recipes/CsvToJsonConversion.cls

```diff
@@ -0,0 +1,45 @@
+/**
+ * @description Demonstrates how to use DataWeave
+ * in Apex to convert CSV to JSON.For more DataWeave for Apex example
+ * head to https://github.com/developerforce/DataWeaveInApex
+ * @group DataWeaveInApex Recipes
+ */
+public with sharing class CsvToJsonConversion {
+    /**
+     * @description Demonstrates how to use DataWeave in Apex
+     * to convert a CSV to a basic JSON.
+     * The class uses the `/dw/csvToJsonBasic.dwl` script to convert CSV to JSON
+     * @param csvPayload CSV data to convert to JSON
+     * @return JSON string
+     */
+    public static String convertCsvToJsonBasic(String csvPayload) {
+        DataWeave.Script script = new DataWeaveScriptResource.csvToJsonBasic();
+        DataWeave.Result result = script.execute(
+            new Map<String, Object>{ 'payload' => csvPayload }
+        );
+        return result.getValueAsString();
+    }
+
+    /**
+     * @description Demonstrates how to use DataWeave script in Apex
+     * to convert a CSV to first JSON and then deserialize it to a list of Salesforce contact records
+     * The class uses the `/dw/csvToJson.dwl` script to convert CSV to JSON
+     * @param data CSV data to convert to Salesforce contact records
+     * @return List of contacts
+     */
+    public static List<Contact> convertCsvToJsonContacts(String data) {
+        DataWeave.Script script = new DataWeaveScriptResource.csvToJson();
+
+        DataWeave.Result result = script.execute(
+            new Map<String, Object>{ 'data' => data }
+        );
+        String jsonResult = result.getValueAsString();
+
+        List<Contact> contacts = (List<Contact>) JSON.deserialize(
```

Comment:
IMO we shouldn't do this double transformation (DW + JSON parsing) as doesn't show the true value of DW: we should use DW to directly turn the CSV to `List<Contact>`.

Reply:
This makes sense! Let me rework this a bit. Thank you

Reply:
@pozil Review feedback addressed! Let me know if you see anything else.